tfjs-converter/src/executor/graph_executor.ts

```diff
@@ -225,12 +264,16 @@ export class GraphExecutor implements FunctionExecutor {
           this.weightMap, tensorArrayMap, tensorListMap,
           this.functionExecutorMap);
       const tensorsMap: NamedTensorsMap = {...this.weightMap};
+      if (this.keepIntermediateTensors) {
+        this.clonedTensorsMap = this.cloneTensorMap(this.weightMap);
+      }
 
       Object.keys(inputs).forEach(name => {
         const [nodeName, index] = parseNodeName(name);
         const tensors: Tensor[] = [];
         tensors[index] = inputs[name];
         tensorsMap[nodeName] = tensors;
+        this.cloneTensorList(tensors, nodeName, this.clonedTensorsMap);
```

Comment:
Tensors should only be cloned if `this.keepIntermediateTensors` is set, right?

Reply:
Oh, I see, `cloneTensorsList` checks if `this.clonedTensorsMap` is null before running. I find that a bit confusing. Can we do the check here instead?

Also, I think it might be better for `this.cloneTensorsList` to return a list of cloned tensors instead of mutating the tensors map. See my nit.